plugins/org.eclipse.n4js/src/org/eclipse/n4js/typesystem/ExpectedTypeJudgment.java

```diff
@@ -346,6 +372,23 @@ public TypeRef caseShiftExpression(ShiftExpression e) {
 		public TypeRef caseRelationalExpression(RelationalExpression e) {
 			switch (e.getOp()) {
 			case INSTANCEOF:
+				EObject rhs = e.getRhs();
+				if (rhs instanceof UnaryExpression) {
+					UnaryExpression ue = (UnaryExpression) rhs;
+					boolean isInv = ue.getOp().equals(UnaryOperator.INV);
```

Comment:
Similar to my comment above: if the operator is not INV, you can skip the rest of the extra handling. You do not even need a local variable then.